File: TP/TP1 - kPPV/kPPV.py
""" programme à compléter du kPPV"""
import csv

nbExParClasse = 50
nbApprent = 25
nbCaract = 4
nbClasse = 3

def lectureFichierCSV():
    with open("iris.data", 'r')as fic:
        lines = csv.reader(fic)
        dataset = list(lines)
    #print(dataset[0], len(dataset))
    for i in range(len(dataset)):
        for j in range(nbCaract):
            dataset[i][j] = float(dataset[i][j])
    #print(dataset[0])

    convertDataList = list()
    convertDataList.append(list())
    convertDataList.append(list())
    convertDataList.append(list())


    for data in dataset:
        if dataset[4] == "Iris-setosa":
            convertDataList[0].append(data[0:4])
        elif (dataset[4] == "Iris-versicolor"):
            convertDataList[1].append(data[0:4])
        else:
            convertDataList[2].append(data[0:4])

    return(convertDataList)


# def calculEucli(data1,data2):
#     dist = sqrt((data1[0]-xb)^2 + (data1-yb)^2 + (data1-zb)^2)
#     return dist

def calculDistances(data, dataset):
    """ retourne les distances entre data et la partie apprentissage de dataset"""

    distances = []
    #
    # distances_eucli = list()
    # distances_eucli.append(list())
    # #
    # for c in convertDataList:
    #     for exemple in c:


	#-------- A faire... --------

    return(distances)

def calculClasse(distances):
    """ retourne le numéro de la classe déterminé à partir des distances """

	#-------- A faire... --------

    return(classe)

if __name__ == "__main__":
    print("Début programme kPPV")
    dataset = lectureFichierCSV()
    print(dataset)

    # Calcule et affiche la matrice de confusion et le taux de reco

	#-------- A faire... --------

#--------------------------------- Fin kPPV -----------------------------------

File: TP/TP1 - kPPV/kPPV.java
package kppv;

import java.io.*;

/**
 * @author hubert.cardot
 */
public class kPPV {
    static int NbEx=50, NbClasse=3, NbCaract=4, NbExApprent=25;
    static Double data[][][] = new Double[NbClasse][NbEx][NbCaract];

    public static void main(String[] args) {
        System.out.println("Début programme kPPV");
        lectureFichier();

        Double X[] = new Double[NbCaract];
        Double Dist[] = new Double[NbClasse*NbExApprent];

        //A faire...

    }
    private static void calculDistances(Double X[], Double Dist[]) {
        //---calcule la distance entre l'ex. X et tous les ex. d'apprentissage

        // A faire...
    }
    private static void lectureFichier() {
        //---lecture des données à partir du fichier iris.data
        String ligne, sousChaine;
        int classe=0, n=0;
        try {
             BufferedReader fic=new BufferedReader(new FileReader("iris.data"));
             while ((ligne=fic.readLine())!=null) {
                for(int i=0;i<NbCaract;i++) {
                    sousChaine = ligne.substring(i*NbCaract, i*NbCaract+3);
                    data[classe][n][i] = Double.parseDouble(sousChaine);
                    System.out.println(data[classe][n][i]+" "+classe+" "+n);
                }
                if (++n==NbEx) { n=0; classe++; }
             }
        }
        catch (Exception e) { System.out.println(e.toString()); }
    }

} //fin classe kPPV
